app/src/main/java/com/dscvit/handly/ui/auth/SigninFragment.kt

```diff
@@ -0,0 +1,137 @@
+package com.dscvit.handly.ui.auth
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import com.dscvit.handly.R
+import com.dscvit.handly.model.Result
+import com.dscvit.handly.model.auth.SigninRequest
+import com.dscvit.handly.ui.PostAuthActivity
+import com.dscvit.handly.util.*
+import com.dscvit.handly.util.PrefHelper.set
+import com.github.ybq.android.spinkit.style.Wave
+import com.onesignal.OneSignal
+import kotlinx.android.synthetic.main.fragment_signin.*
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+
+
+class SigninFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_signin, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val authViewModel by sharedViewModel<AuthViewModel>()
+        val sharedPref = PrefHelper.customPrefs(requireContext(), Constants.PREF_NAME)
+
+        val action = SigninFragmentDirections.actionSigninFragmentToSignupFragment()
+        signin_progress.hide()
+        signin_progress.setIndeterminateDrawable(Wave())
+
+        signup_instead_text.setOnClickListener {
+            findNavController().navigate(action)
+        }
+
+        signin_button.setOnClickListener {
+            val email = signin_email_text.editText!!.text.toString()
+            val password = signin_password_text.editText!!.text.toString()
+
+            if (!email.isValidEmail()) {
+                signin_email_text.error = "Please enter a valid email"
```

Comment:
Consider extracting all strings to resources

Reply:
👍🏼